include/dem/velocity_verlet_integrator.h

```diff
@@ -102,8 +102,18 @@ class VelocityVerletIntegrator : public Integrator<dim>
             std::vector<Tensor<1, 3>>                       &force,
             const std::vector<double>                       &MOI,
             const parallel::distributed::Triangulation<dim> &triangulation,
-            typename DEM::dem_data_structures<dim>::cell_index_int_map
-              &cell_mobility_status_map) override;
+            AdaptiveSparseContacts<dim> &sparse_contacts_object) override;
+
+  void
+  integrate_with_advected_particles(
+    Particles::ParticleHandler<dim>                 &particle_handler,
+    const Tensor<1, 3>                              &body_force,
+    const double                                     time_step,
+    std::vector<Tensor<1, 3>>                       &torque,
+    std::vector<Tensor<1, 3>>                       &force,
+    const std::vector<double>                       &MOI,
+    const parallel::distributed::Triangulation<dim> &triangulation,
+    AdaptiveSparseContacts<dim>                     &sparse_contacts_object);
```

Comment:
I know it sucks to ask for you to document the function because the one above is not, but if you feel like you have the time :).

Reply:
Can do it at the end of the reviews

Reply:
Actually, I feel it's better to do it in a refactoring PR itself, otherwise it adds lots of random changes to a PR and this PR is already huge enough for reviewers in my opinion. Otherwise the focus won't be on the main things.

Reply:
Agreed
Good judgment call here